Submodels/MTSOL_call.py

```
@@ -198,34 +303,77 @@ def GenerateProcess(self,
+                try:
+                    self.process.wait(timeout=10)  # Wait for the process to terminate after killing
+                except subprocess.TimeoutExpired:
+                    print("Warning: process could not be terminated within timeout")
+
+        self.process = subprocess.Popen([self.fpath, self.analysis_name],
+                                        stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.DEVNULL,
                                         text=True,
                                         bufsize=1,
                                         )
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)